posthog/hogql_queries/web_analytics/test/test_revenue.py

```diff
@@ -157,3 +185,87 @@ def test_revenue_currency_static(self):
         assert isinstance(response, CachedTrendsQueryResponse)
         results = response.results[0]
         assert results["data"] == [Decimal("318.3902190523"), Decimal("569.764497341")]
+
+    def test_dw_revenue_currency_property(self):
+        self._create_data_warehouse_table("database_with_revenue_column", "id", "revenue", "timestamp", "currency")
+
+        # Spy on the `clichhouse_driver.Client.execute` method to avoid querying the data warehouse tables
```

Comment:
Ooof. This is indeed hacky. Impressive! But hacky. I'm ok with this for now but this should be on the list of things to improve. Maybe @PostHog/team-data-warehouse can help? I wonder if we could make a clickhouse <- postgres bridge the test-case, rather than a clickhouse <- s3 bridge, so that we don't need to mock it

Reply:
I **think** I can make this work with a mock CSV file instead. l now feel justified to spend some time to try and improve DX now that revenue analytics is its own product :)